src/utils/utils.js

```diff
@@ -26,3 +37,59 @@ export function createNewDayCells() {
   }
   return ROW_DATA;
 }
+export function calculateSummaryOfSleep(cells) {
+  const reducer = (map, val) => {
+    if (map[val] == null) {
+      map[val] = 1;
+    } else {
+      ++map[val];
+    }
+    return map;
+  };
+  const result = cells.map((cell) => cell.mode).reduce(reducer, {});
+  return result;
+}
+export function clusterCells(cells) {
+  let currentMode = null;
+  let obj = {
+    startIndex: null,
+    endIndex: null,
+    mode: null
+  };
+  const clusteredCells = cells.reduce((result, current, i) => {
+    if (currentMode !== current.mode && currentMode != null) {
+      obj = { ...obj, endIndex: i - 1 };
+      result.push(obj);
+      currentMode = current.mode;
+      obj = {
+        endIndex: null,
+        startIndex: i,
+        mode: current.mode
+      };
+    }
+    if (i === cells.length - 1) {
+      obj = { ...obj, endIndex: i };
+      result.push(obj);
+    }
+    if (currentMode === null) {
+      currentMode = current.mode;
+      obj = { ...obj, startIndex: i, mode: current.mode };
+    }
+
+    return result;
+  }, []);
+
+  return clusteredCells;
```

Comment:
bug: Take a look at [this scenario](https://user-images.githubusercontent.com/14869087/164108631-eff68bc9-5af3-4487-94b7-0cec14f05abc.png) — you'll notice some gaps between blockers that are bigger than expected. You can reproduce it by recording an "active" block of just 15min.

Do you understand why? How would you solve it?

Reply:
Perhaps we should show only 3 clusters, summing the total of each mode... But I let you decide, which one do you prefer the most?